react/features/toolbox/components/web/Toolbox.js

```diff
@@ -1133,27 +1151,89 @@ function _mapStateToProps(state) {
     } = state['features/toolbox'];
     const localParticipant = getLocalParticipant(state);
     const localVideo = getLocalVideoTrack(state['features/base/tracks']);
-    const isModerator = localParticipant.role === PARTICIPANT_ROLE.MODERATOR;
     const addPeopleEnabled = isAddPeopleEnabled(state);
     const dialOutEnabled = isDialOutEnabled(state);
+    let recordingEnabled
+        = isLocalParticipantModerator(state) && fileRecordingsEnabled;
+    let streamingEnabled
+        = isLocalParticipantModerator(state) && liveStreamingEnabled;
+
+    let desktopSharingDisabledTooltipKey;
+    let fileRecordingsDisabledTooltipKey;
+    let liveStreamingDisabledTooltipKey;
+
+    if (state['features/base/config'].enableFeaturesBasedOnToken) {
+        // we enable desktop sharing if any participant already have this
+        // feature enabled
+        desktopSharingEnabled = getParticipants(state)
+            .find(({ features = {} }) =>
+                String(features['screen-sharing']) === 'true') !== undefined;
+
+        // we want to show button and tooltip
+        if (state['features/base/jwt'].isGuest) {
+            desktopSharingDisabledTooltipKey
+                = 'dialog.shareYourScreenDisabledForGuest';
+        } else {
+            desktopSharingDisabledTooltipKey
+                = 'dialog.shareYourScreenDisabled';
+        }
+
+        // we enable recording if the local participant have this
+        // feature enabled
+        const { features = {} } = localParticipant;
+        const { isGuest } = state['features/base/jwt'];
+
+        recordingEnabled
```

Comment:
I think you should rename this variable to fileRecordingEnabled or something that refers to this as file. And streamingEnabled should reference live streaming, if possible. That way there's more consistency with the naming. We/I screwed myself when recording was overloaded for live streaming and file recording and we should try to proceed by making the separation between the two more clear.